src/main/java/walter/Walter.java

```diff
@@ -0,0 +1,443 @@
+        fwClear.write("");
+        fwClear.close();
+
+        //Append information into file
+        FileWriter fw = new FileWriter(FILE_PATH, true);
+        for (Task task : tasks) {
+            int taskStatus;
+            //Determine status to write to file based on task status
+            if (task.getStatus()) {
+                taskStatus = 1;
+            } else {
+                taskStatus = 0;
+            }
+            String taskToSave = task.getTaskIcon() + SAVE_DELIMITER + taskStatus + SAVE_DELIMITER
+                    + task.getDescription() + SAVE_DELIMITER + task.getTimingInformation() + System.lineSeparator();
+            fw.write(taskToSave);
+        }
+        fw.close();
+    }
+
+    /**
+     * Recreates tasks array by reading data from file. If no file available, create new file
+     */
+    public static void readFileContents(ArrayList<Task> tasks) throws IOException {
+        File f = new File(FILE_PATH);
+
+        //Read from file if exists, else create new directory and files
+        if (f.exists()) {
+            Scanner s = new Scanner(f);
+            //Re-create task objects in the array
+            while (s.hasNext()) {
+                String taskInformation = s.nextLine();
+                String[] taskComponents = taskInformation.split(SAVE_DELIMITER);
+                switch (taskComponents[0]) {
+                case TODO_ICON:
+                    tasks.add(new Todo(taskComponents[2]));
+                    break;
+                case DEADLINE_ICON:
+                    tasks.add(new Deadline(taskComponents[2], taskComponents[3]));
+                    break;
+                case EVENT_ICON:
+                    tasks.add(new Event(taskComponents[2], taskComponents[3]));
+                    break;
+                }
+                //Set status of task to done if required
+                if (Integer.parseInt(taskComponents[1]) == 1) {
+                    tasks.get(tasks.size() - 1).setAsDone();
+                }
+            }
+        } else {
+            //No existing file detected. Create new save file
+            System.out.println(FILE_MESSAGE_NO_SAVE_DETECTED);
+            boolean fileCreated = f.createNewFile();
+            if (fileCreated) {
+                System.out.println(FILE_MESSAGE_CREATED_SUCCESS);
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        //Initialise variables
+        String userInput;
+        String[] splitUserInput;
+        ArrayList<Task> tasks = new ArrayList<>();
+        boolean isFinished = true;
+        Scanner in = new Scanner(System.in);
+
+        //Print startup sequence
+        printStartupSequence();
+
+        //Read information from saved files
+        try {
+            readFileContents(tasks);
+        } catch (IOException e) {
+            System.out.println(EXCEPTION_FILE_ERROR);
+        }
+
+        //Loop infinitely until user enters "bye"
+        while (isFinished) {
```

Comment:
Consider initializing isFinished to false instead and changing the loop condition to while (!isFinished), to make it easier to understand.